common/app/views/fragments/preRenderJavaScript.scala.html

```diff
@@ -0,0 +1,58 @@
+@()
+
+@* This script must come after the pre-flight shims *@
+
+<script>
+(function applyAdFree(){
+    /**
+     * Determine if user has 'adfree' view, and apply appropriate body classes before rendering.
+     */
+
+    var ADFREE_COOKIE_KEY = 'gu_adfree_user',
+        USER_SIGNIN_COOKIE = 'GU_U',
+        commercial = guardian.config.commercial;
+
+    if (showAdfreeView()) {
+        addClass(document.documentElement, 'adfree');
+        commercial.showingAdfree = true;
+    } else {
+        commercial.showingAdfree = false;
+    }
+
+    function showAdfreeView() {
+        if (featureDisabled() || userLoggedOut()) {
+            return false;
+        } else {
+            var adfreeUser = readCookie(ADFREE_COOKIE_KEY);
+            // If the cookie is ambiguous or null, err on the side of caution and hold off showing ads
+            return adfreeUser !== 'false';
+        }
+    }
+
+    function featureDisabled() {
+        return guardian.config.switches.advertOptOut === false;
```

Comment:
yeah sorry, maybe i picked the wrong place for this. i mean currently this code defines some variables and functions and calls them before we really know we need to.

if `guardian.config.switches.advertOptOut === false`, for example,  we can return from the iffe immediately, is that right? 

only because this code is run immediately, everywhere, i think it's good to make it do the absolute bare minimum. outside of such crucial stuff, micro-optimisations seem a bit naval-gazey; but in this, i think it would be prudent to make them...?